### ISSUE TITLE
MixedPlaySpec runs tests in Parallel

### ISSUE THREAD

**Issue by cheeseng** [playframework/scalatestplus-play#35]
This was originally reported in the following playframework PR: 

https://github.com/playframework/playframework/pull/5514

It was reported that MixedPlaySpec runs its tests parallel unexpectedly.

Here's the examples given by @schmitch: 

This one will keep my database connections at a peek of 21:

``` scala
class AppDoubleSpec extends PlaySpec with OneAppPerTest{

  var mutSeq = mutable.ListBuffer[String]()

  "Blabla" should {
    "be only happen once first" in {
      mutSeq.length must be(0)
    }
    "be only happen once second" in {
      mutSeq += "hello world"
      mutSeq.length must be(1)
    }
    "be only happen once third" in {
      mutSeq.length must be(1)
    }
  }

  "Blabla2" should {
    "be only happen once" in {
      mutSeq.length must be(1)
    }
  }

}
```

This one will peek my connections to 84:

``` scala
class AppDoubleSpec extends MixedPlaySpec {

  var mutSeq = mutable.ListBuffer[String]()

  "Blabla" should {
    "be only happen once first" in new App {
      mutSeq.length must be(0)
    }
    "be only happen once second" in new App {
      mutSeq += "hello world"
      mutSeq.length must be(1)
    }
    "be only happen once third" in new App {
      mutSeq.length must be(1)
    }
  }

  "Blabla2" should {
    "be only happen once" in new App {
      mutSeq.length must be(1)
    }
  }

}
```

and he is using the following python script to look at the opened connection: 

``` python
import psycopg2

conn = psycopg2.connect("dbname=loki user=loki password=loki")
conn.autocommit = True

last = None
with conn:
    with conn.cursor() as cur:
        while True:

            ret = cur.execute("SELECT sum(numbackends) FROM pg_stat_database;")
            current = cur.fetchone()[0] - 1

            if current != last and current - 1 != last:
                print(current)

            last = current

conn.close()
```

and @bvenners mentioned this: 

> The one difference I can see is that we are still using FakeApplication in MixedFixtures, but are using GuicrApplicationBuilder on OneSppPerTest. That's unintended anyway because we thought we'd completely switched over to GuiceApplicationBuilder. We'll fix that, but I'd like to try and reproduce your issue. How are you counting your database connections? (Probably we should move this to a ScalaTest+Play issue.)

**Comment by schmitch** [comment]
As a note I don't think switching to GuiceApplicationBuilder will fix the issue since I discovered the issue with a GuiceApplicationBuilder I've wanted to run tests with different data.
But actually I'm not sure if my script was the problem.
Here's how I've done it:
```
abstract class EnvisiaMixedSpec extends MixedPlaySpec with EnvisiaApplicationSpec with DBIOAwaits {

  abstract class Database(builder: GuiceApplicationBuilder = builder) extends App(builder.build()) with EnvisiaFixtures {

    override def apply() {
      val dbConfigProvider = app.injector.instanceOf[DatabaseConfigProvider]
      val database = app.injector.instanceOf[DBApi].database("default")

      val dbConfig = dbConfigProvider.get[PostgresDriver]
      val driver = dbConfig.driver
      val db = dbConfig.db
      import driver.api._
      val duration = Duration.create(10, TimeUnit.SECONDS)
      Await.result(db.run(sql"DROP SCHEMA IF EXISTS public CASCADE".asUpdate), duration)
      Await.result(db.run(sql"CREATE SCHEMA public".asUpdate), duration)
      Evolutions.applyEvolutions(database)

      // Applies each fixture
      val autocommit = false
      database.withConnection(autocommit) { implicit c =>
        fixtures.foreach { file =>
          FixtureLoader.apply(file, autocommit)
        }
      }
      super.apply()
    }
  }

}
```

**Comment by cheeseng** [comment]
@schmitch @bvenners I am looking into this and I think I have found the problem, it seems that the Application passed to new App (implicitly created FakeApplication instance) was not lazy, making it lazy seems to fix the problem for me, I shall be able to submit a PR for 1.4.x branch today.

Thanks.

**Comment by cheeseng** [comment]
@schmitch @bvenners PR for 1.4.x is here: 

https://github.com/playframework/scalatestplus-play/pull/36

If it looks good I think we should back port it to 1.0.x -> 1.3.x series, I will make one for master (for Play 2.5) first in case @jroper want to release a RC.

**Comment by schmitch** [comment]
@cheeseng thanks, now everything works. didn't knew that it's so 'simple' to fix.
I'm not sure about the backport, doesn't look that many people found that bug.

**Comment by bvenners** [comment]
@cheeseng Thanks for tracking it down. Makes sense now. I checked over both PRs and they both look good. I'll start the testing process.

**Comment by schmitch** [comment]
@cheeseng could you close this ? I think you fixed that in #36

**Comment by marcospereira** [comment]
Closing since this was fixed by #36.
